File: backend/app/api/schemas/strategy.py
import uuid

from pydantic import BaseModel

from app.domain.enums.strategy_action_type import StrategyActionType


class StrategyActionOut(BaseModel):
    action_type: StrategyActionType
    reason: str
    priority: int
    application_id: uuid.UUID | None = None
    offer_id: uuid.UUID | None = None


class PrioritizedOfferOut(BaseModel):
    offer_id: uuid.UUID
    title: str
    priority_score: float
    ranking_score: float | None
    matching_score: float | None
    location_text: str | None = None
    company_name: str | None = None


class SkillGapOut(BaseModel):
    offer_id: uuid.UUID
    offer_title: str
    missing_skills: list[str]


class StrategyRecommendationsOut(BaseModel):
    actions: list[StrategyActionOut]
    prioritized_offers: list[PrioritizedOfferOut]
    skill_gaps: list[SkillGapOut]

File: frontend/src/types/strategy.ts
export type StrategyActionType =
  | "APPLY_NOW"
  | "SEND_FOLLOWUP"
  | "PREPARE_INTERVIEW"
  | "ARCHIVE_STALE";

export interface StrategyAction {
  action_type: StrategyActionType;
  reason: string;
  priority: number;
  application_id: string | null;
  offer_id: string | null;
}

export interface PrioritizedOffer {
  offer_id: string;
  title: string;
  priority_score: number;
  ranking_score: number | null;
  matching_score: number | null;
  location_text: string | null;
  company_name: string | null;
}

export interface SkillGap {
  offer_id: string;
  offer_title: string;
  missing_skills: string[];
}

export interface StrategyRecommendations {
  actions: StrategyAction[];
  prioritized_offers: PrioritizedOffer[];
  skill_gaps: SkillGap[];
}
